graphql/resolvers.go

```diff
@@ -235,6 +230,49 @@ func (r *mutationResolver) SpawnVolume(ctx context.Context, spawnVolumeInput Spa
 	return success, nil
 }
 
+func (r *mutationResolver) UpdateVolume(ctx context.Context, updateVolumeInput UpdateVolumeInput) (bool, error) {
+	volume, err := r.sc.FindVolumeById(updateVolumeInput.VolumeID)
+	if err != nil {
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("Error finding volume by id %s: %s", updateVolumeInput.VolumeID, err.Error()))
+	}
+	if volume == nil {
+		return false, ResourceNotFound.Send(ctx, fmt.Sprintf("Unable to find volume %s", volume.ID))
+	}
+	err = validateVolumeExpirationInput(ctx, updateVolumeInput.Expiration, updateVolumeInput.NoExpiration)
+	if err != nil {
+		return false, err
+	}
+	err = validateVolumeName(ctx, updateVolumeInput.Name)
+	if err != nil {
+		return false, err
+	}
+	var updateOptions restModel.VolumeModifyOptions
+	if updateVolumeInput.NoExpiration != nil && *updateVolumeInput.NoExpiration == true {
+		// this value should only ever be true or nil
+		updateOptions.NoExpiration = true
+	}
+	if updateVolumeInput.NoExpiration != nil && *updateVolumeInput.NoExpiration == false {
```

Comment:
you can make this a bit cleaner by doing
if (noExpiration != nil) {
  if (noExpiration)...
  else...
}